src/Stratis.Bitcoin.Features.Consensus/ProvenBlockHeaders/ProvenBlockHeaderStore.cs

```diff
@@ -288,43 +310,134 @@ private async Task SaveAsync()
                 this.pendingTipHashHeight = null;
             }
 
-            this.CheckItemsAreInSequence(pendingBatch.Keys.ToList());
+            this.CheckItemsAreInConsecutiveSequence(pendingBatch.Keys.ToList());
 
+            // Save the items to disk.
             using (new StopwatchDisposable(o => this.performanceCounter.AddInsertTime(o)))
             {
                 // Save the items to disk.
-                await this.provenBlockHeaderRepository.PutAsync(pendingBatch.Values.ToList(), hashHeight);
+                await this.provenBlockHeaderRepository.PutAsync(pendingBatch.Values.ToList(), hashHeight).ConfigureAwait(false);
+
+                this.TipHashHeight = this.provenBlockHeaderRepository.TipHashHeight;
+            }
+        }
+
+        /// <summary>
+        /// Will recover the <see cref="IProvenBlockHeaderStore"/> tip to the most recent <see cref="ProvenBlockHeader"/> 
+        /// that exists in the <see cref="IProvenBlockHeaderRepository"/> and/or <see cref="ChainedHeader"/>.
```

Comment:
item that exists in repo and chained header? 
Did you mean and is ancestor of provided chained header?

Reply:
I wasn't too sure how to describe what I was using from the `ChainHeader`.  I'll use your recommend description.